controllers/iamrole_controller.go

```diff
@@ -72,43 +73,43 @@ func (r *IamroleReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.WithValue(context.Background(), requestId, uuid.New())
 	log := log.Logger(ctx, "controllers", "iamrole_controller", "Reconcile")
 	log.WithValues("iamrole", req.NamespacedName)
-	log.Info("Start of the request")
+	log.Info("Start of the request", "resource", req)
 
-	//Get the resource
+	// Get the resource and (if debugging is on) print out the current state as it sits in K8S
 	var iamRole iammanagerv1alpha1.Iamrole
-
 	if err := r.Get(ctx, req.NamespacedName, &iamRole); err != nil {
 		return ctrl.Result{}, ignoreNotFound(err)
 	}
+	log.V(1).Info("Current state of the resource", "iamrole", iamRole)
 
 	// Is it being deleted?
 	if iamRole.ObjectMeta.DeletionTimestamp.IsZero() {
-		//Good. This is not Delete use case
-		//Lets check if this is very first time use case
+		// Good. This is not Delete use case
+		// Lets check if this is very first time use case
 		if !validation.ContainsString(iamRole.ObjectMeta.Finalizers, finalizerName) {
 			log.Info("New iamrole resource. Adding the finalizer", "finalizer", finalizerName)
 			iamRole.ObjectMeta.Finalizers = append(iamRole.ObjectMeta.Finalizers, finalizerName)
 			r.UpdateMeta(ctx, &iamRole)
 		}
 		return r.HandleReconcile(ctx, req, &iamRole)
-
 	} else {
-		//oh oh.. This is delete use case
-		//Lets make sure to clean up the iam role
+		// oh oh.. This is delete use case
+		// Lets make sure to clean up the iam role
 		if iamRole.Status.RetryCount != 0 {
 			iamRole.Status.RetryCount = iamRole.Status.RetryCount + 1
 		}
 		log.Info("Iamrole delete request")
 		if iamRole.Status.State != iammanagerv1alpha1.PolicyNotAllowed {
 			//Get the roleName from status
 			roleName := iamRole.Status.RoleName
-			if err := r.IAMClient.DeleteRole(ctx, roleName); err != nil {
-				log.Error(err, "Unable to delete the role")
-				//i got to fix this
-				r.UpdateStatus(ctx, &iamRole, iammanagerv1alpha1.IamroleStatus{RetryCount: iamRole.Status.RetryCount + 1, LastUpdatedTimestamp: metav1.Now(), ErrorDescription: err.Error(), State: iammanagerv1alpha1.Error})
-				r.Recorder.Event(&iamRole, v1.EventTypeWarning, string(iammanagerv1alpha1.Error), "unable to delete the role due to "+err.Error())
-
-				return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
+			if roleName != "" {
```

Comment:
Now, That i'm checking more on IRSA- we might have to clean up the service account too to make sure we no longer have "deleted iam role" annotation to the service account